frontend/lib/resolution/scope-queries.cpp

```diff
@@ -486,17 +504,51 @@ static bool doLookupInScope(Context* context,
     // Search parent scopes, if any, until a module is encountered
     const Scope* cur = nullptr;
     bool reachedModule = false;
+    bool skipClosestConditional = false;
+
+    // This trickiness is required to implement correct scoping behavior
+    // for 'if-vars' in conditionals. The 'if-var' lives in the scope
+    // for the conditional, but it is not visible within the 'else'
+    // branch. Without this hack, we'd be able to see the 'if-var' in
+    // both branches. First, detect if the start scope is the else-block.
+    //
+    // TODO: Consider query-caching some part of this pattern matching.
+    if (!scope->id().isEmpty())
+      if (auto ast = parsing::idToAst(context, scope->id()))
+        if (isElseBlockOfConditionalWithIfVar(context, ast))
+          skipClosestConditional = true;
+
     for (cur = scope->parentScope(); cur != nullptr; cur = cur->parentScope()) {
       if (asttags::isModule(cur->tag())) {
         reachedModule = true;
         break;
       }
 
+      auto ast = !cur->id().isEmpty() ? parsing::idToAst(context, cur->id())
+                                      : nullptr;
+
+      // We could be in a nested block, so check for the else-block to
+      // trigger the pattern matching as we walk up...
+      if (!skipClosestConditional && ast)
+        if (isElseBlockOfConditionalWithIfVar(context, ast))
+          skipClosestConditional = true;
+
+      // Skip the first conditional's scope if we need to.
+      if (skipClosestConditional) {
+        if (ast && ast->isConditional()) {
+          skipClosestConditional = false;
+          continue;
+        }
+      }
```

Comment:
Should these two if-blocks be swapped? In a nested if-var-else case like this:

```Chapel
if var x = f() {}
else if var x = g() {}
else x;
```

The current flow is:
1. On line 3, set `skipClosestConditional`. Go up one scope.
2. On line 2, (inner if), set `skipClosestConditional` again 
3. On line 2, (inner if), skip the conditional and reset `skipClosestConditional`
4. On line 1, (outer if), lookup x.

Now that I think about it, there's always an intermediate `Block` in there so the two conditions should never occur at the same time, so maybe this question is moot.

Reply:
The above might be a good sort of test to add, anyway.